doc/src/tutorials/creating-packages.md

```diff
@@ -292,22 +296,23 @@ Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 test = ["Markdown", "Test"]
 ```
 
+During testing the file `test/runtests.jl` will be executed with all dependencies from the `Project.toml` and the test dependencies
+loaded. This special dependency handling is done during a `Pkg.test` call and doesn't happen if `runtest.jl` is executed directly.
+
 Note that the only supported targets are `test` and `build`, the latter of which (not recommended) can be used
-for any `deps/build.jl` scripts.
+for any `deps/build.jl` scripts. This `target` based approach can't be used for other uses such as building
+documentation.
 
 #### Alternative approach: `test/Project.toml` file test specific dependencies
 
-!!! note
-    The exact interaction between `Project.toml`, `test/Project.toml` and their corresponding
-    `Manifest.toml`s are not fully worked out and may be subject to change in future versions.
-    The older method of adding test-specific dependencies, described in the previous section,
-    will therefore be supported throughout all Julia 1.X releases.
+In Julia 1.2 and later test dependencies can be declared in `test/Project.toml`. This is similar
```

Comment:
```suggestion
!!! compat
    This method of adding test-specific dependencies is supported for all Julia 1.x releases since v1.2.
    
Test dependencies can alternatively be declared in `test/Project.toml`. This is similar
```